src/components/submission/filters/submitter.vue

```diff
@@ -11,15 +11,17 @@ except according to the terms contained in the LICENSE file.
 -->
 <template>
   <label id="submission-filters-submitter" class="form-group">
-    <select class="form-control" :value="value"
+    <select v-if="initiallyLoading" class="form-control" disabled>
+      <option>{{ $t('common.loading') }}</option>
+    </select>
+    <select v-else class="form-control" :value="value"
```

Comment:
@ktuite, the problem here turned out to be different from what I thought. My thought had been that `:value="value"` was setting the value attribute of the select element (corresponding to the element's initial value) instead of its value property (corresponding to its current selected value). However, I think I was wrong about that: `:value="value"` does seem to set the `value` property. For example, I added this test, and it passed without changes to the component:

https://github.com/getodk/central-frontend/blob/baa3f22c6ed98c40ffd8d61aeed251d25c182caa/test/components/submission/filters/submitter.spec.js#L89-L94

However, this similar test failed:

https://github.com/getodk/central-frontend/blob/baa3f22c6ed98c40ffd8d61aeed251d25c182caa/test/components/submission/filters/submitter.spec.js#L96-L101

As did a test representing the bug we observed:

https://github.com/getodk/central-frontend/blob/baa3f22c6ed98c40ffd8d61aeed251d25c182caa/test/components/submission/filters/submitter.spec.js#L65-L82

I'm thinking now that the problem had to do with the fact that the options are dynamic. In the second test above, the "Unknown submitter" option is added once the `value` prop is changed to `'42'`. However, I'm thinking that Vue might be setting the `value` property of the select element before inserting the new option. And since the option doesn't exist yet, setting the value fails, and the browser falls back to the first option.

That would also explain the behavior of the third test, because again, new options will be inserted after `submitters` has loaded.

I tried setting the `key` attribute on the select element (in the `v-if` and the `v-else`). That caused the third test to pass, but the second test still failed. (By the key, this behavior of `key` with `v-if` is [going to change](https://v3-migration.vuejs.org/breaking-changes/key-attribute.html) in Vue 3 in a way that I think is nicer/simpler.)

I then tried `v-model` with a computed property (a pattern that I've used elsewhere but that I still find a little awkward). Happily, that was successful.

With these tests in place, I feel confident that things are working as they should. It's definitely an interesting case! Please feel free to take a look at that commit and/or to try out these filters and see whether they work as you'd expect.